File: aladdinworksfivefiftyfive_ui/src/services/EnvironmentalMetricService.js
import http from "../http-common"; 

class EnvironmentalMetricService {
  getAllEnvironmentalMetrics(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/environmentalMetric/environmentalMetrics`, searchDTO);
  }

  get(environmentalMetricId) {
    return this.getRequest(`/environmentalMetric/${environmentalMetricId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/environmentalMetric?field=${matchData}`, null);
  }

  addEnvironmentalMetric(data) {
    return http.post("/environmentalMetric/addEnvironmentalMetric", data);
  }

  update(data) {
  	return http.post("/environmentalMetric/updateEnvironmentalMetric", data);
  }
  
  uploadImage(data,environmentalMetricId) {
  	return http.postForm("/environmentalMetric/uploadImage/"+environmentalMetricId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new EnvironmentalMetricService();

File: src/main/java/com/aladdinworksfivefiftyfive/service/EnvironmentalMetricService.java
package com.aladdinworksfivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworksfivefiftyfive.domain.EnvironmentalMetric;
import com.aladdinworksfivefiftyfive.dto.EnvironmentalMetricDTO;
import com.aladdinworksfivefiftyfive.dto.EnvironmentalMetricSearchDTO;
import com.aladdinworksfivefiftyfive.dto.EnvironmentalMetricPageDTO;
import com.aladdinworksfivefiftyfive.dto.EnvironmentalMetricConvertCriteriaDTO;
import com.aladdinworksfivefiftyfive.service.GenericService;
import com.aladdinworksfivefiftyfive.dto.common.RequestDTO;
import com.aladdinworksfivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface EnvironmentalMetricService extends GenericService<EnvironmentalMetric, Integer> {

	List<EnvironmentalMetric> findAll();

	ResultDTO addEnvironmentalMetric(EnvironmentalMetricDTO environmentalMetricDTO, RequestDTO requestDTO);

	ResultDTO updateEnvironmentalMetric(EnvironmentalMetricDTO environmentalMetricDTO, RequestDTO requestDTO);

    Page<EnvironmentalMetric> getAllEnvironmentalMetrics(Pageable pageable);

    Page<EnvironmentalMetric> getAllEnvironmentalMetrics(Specification<EnvironmentalMetric> spec, Pageable pageable);

	ResponseEntity<EnvironmentalMetricPageDTO> getEnvironmentalMetrics(EnvironmentalMetricSearchDTO environmentalMetricSearchDTO);
	
	List<EnvironmentalMetricDTO> convertEnvironmentalMetricsToEnvironmentalMetricDTOs(List<EnvironmentalMetric> environmentalMetrics, EnvironmentalMetricConvertCriteriaDTO convertCriteria);

	EnvironmentalMetricDTO getEnvironmentalMetricDTOById(Integer environmentalMetricId);







}





